modules/latex.php

```diff
@@ -57,28 +216,117 @@ function latex_src( $matches ) {
 	return latex_render( $latex, $fg, $bg, $s );
 }
 
+/**
+ * Get colors for LaTeX expressions.
+ *
+ * @since 1.1
+ *
+ * @global array $themecolors
+ *
+ * @param  string $color
+ * @param  string $default_color Default 'ffffff'.
+ * @return string
+ */
 function latex_get_default_color( $color, $default_color = 'ffffff' ) {
 	global $themecolors;
 	return isset($themecolors[$color]) ? $themecolors[$color] : $default_color;
 }
 
+/**
+ * Decode HTML entities in LaTeX expressions.
+ *
+ * @since 1.1
+ *
+ * @param  string $latex LaTeX expression.
+ * @return string        LaTeX expression.
+ */
 function latex_entity_decode( $latex ) {
 	return str_replace( array( '&lt;', '&gt;', '&quot;', '&#039;', '&#038;', '&amp;', "\n", "\r" ), array( '<', '>', '"', "'", '&', '&', ' ', ' ' ), $latex );
 }
 
+/**
+ * Convert latex shortcode size into a valid LaTeX size command.
+ *
+ * @since 3.8
+ *
+ * @param  int $s
+ * @return mixed  LaTeX size command if successful, otherwise false.
+ */
+function latex_size_it( $s ) {
+	switch ( (int) $s ) {
+	case 0 :
+		return false;
+	case 1 :
+		return '\large';
+	case 2 :
+		return '\Large';
+	case 3 :
+		return '\LARGE';
+	case 4 :
+		return '\huge';
+	case 5 :
+		return '\Huge';
+	case -1 :
+		return '\small';
+	case -2 :
+		return '\footnotesize';
+	case -3 :
+		return '\scriptsize';
+	case -4 :
+		return '\tiny';
+	default :
+		return false;
+	}
+}
+
+/**
+ * Print a valid MathJax expression.
+ *
+ * Wraps LaTeX expression with LaTeX font size command sequence when necessary.
+ * Wraps expression in a span to control colors when necessary.
+ *
+ * @since 1.1
+ *
+ * @param  string  $latex LaTeX expression.
+ * @param  string  $fg    Foreground color.
+ * @param  string  $bg    Background color.
+ * @param  integer $s     Font size. Default '0'.
+ * @return string         LaTeX expression.
+ */
 function latex_render( $latex, $fg, $bg, $s = 0 ) {
-	$url = "//s0.wp.com/latex.php?latex=" . urlencode( $latex ) . "&bg=" . $bg . "&fg=" . $fg . "&s=" . $s;
-	$url = esc_url( $url );
-	$alt = str_replace( '\\', '&#92;', esc_attr( $latex ) );
+	$latex_size = latex_size_it( $s );
+	$styles      = array();
+
+	if ( $fg ) {
+		$styles[] = 'color: #' . $fg . ';';
+	}
+	if ( $bg ) {
+		$styles[] = 'background-color: #' . $bg . ';';
+	}
+
+	if ( $latex_size ) {
+		// Add LaTeX code to a font size command.
+		$latex = sprintf( $latex_size . '{%s}', $latex );
+	}
 
-	return '<img src="' . $url . '" alt="' . $alt . '" title="' . $alt . '" class="latex" />';
+	if ( $styles ) {
+		return '<span style="display: inline-block;' . implode('', $styles) . '">\(' . $latex . '\)</span>';
+	} else {
+		return '\(' . $latex . '\)';
```

Comment:
:+1: 

If we do get rid of `\( .. )\` so we don't break any old posts this will become:

```php
'inlineMath' => array(
	array( '$latex ', '$' ),
),
```